src/code/views/node-view.coffee

```diff
@@ -16,6 +16,11 @@ NodeTitle = React.createFactory React.createClass
     if @props.isEditing
       @inputElm().off()
 
+  componentWillUpdate: (nextProps) ->
+    # mark the title as updated even if no change was made when it leaves edit mode
+    @props.node.titleUpdated = true if @props.isEditing and not nextProps.isEditing
+    @canDeleteWhenEmpty = @props.node.addedThisSession and not @props.node.titleUpdated
```

Comment:
Another quick question just to understand this better: why have `@canDeleteWhenEmpty` be a class variable, but have `titleUpdated` be a prop? `titleUpdated` could just be a variable, right, since we only care about it during the lifecycle of this view?